src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunHelperNode.cs

```diff
@@ -162,6 +162,24 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                 dependencyList.Add(factory.TypeThreadStaticsSymbol((MetadataType)_target), "ReadyToRun Thread Static Storage");
                 return dependencyList;
             }
+            else if (_id == ReadyToRunHelperId.ResolveGenericVirtualMethod)
+            {
+                MethodDesc method = _target as MethodDesc;
+                Debug.Assert(method != null && method.HasInstantiation && method.IsVirtual);
+
+                DependencyList dependencyList = new DependencyList();
+
+                // If the target method is an exact (non-canonical) instantiation, we'll need an entry for it in the exact method instantiations 
+                // hashtable, so we append the entry's dependencies to the method here.
+                var exactMethodInstantiationDependencies = ExactMethodInstantiationsNode.GetExactMethodInstantiationDependenciesForMethod(factory, method);
```

Comment:
How does an entry for this end up in the `ExactMethodInstantiations` hashtable though? It seems like we generate dependencies for a thing that never actually lands in the hashtable.